modules/swagger-codegen/src/main/resources/JavaSpring/beanValidation.mustache

```diff
@@ -1,5 +1,6 @@
 {{#required}}
   @NotNull
-{{/required}}
+{{/required}}{{#isContainer}}{{^isPrimitiveType}}{{^isEnum}}
```

Comment:
@wing328 on my part I can safely say that it works as intended since we have deployed this fix in production for a while now :)